src/mlpack/methods/matrix_completion/matrix_completion_impl.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file matrix_completion_impl.hpp
+ * @author Stephen Tu
+ *
+ * Implementation of MatrixCompletion class.
+ */
+#ifndef __MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_IMPL_HPP
+#define __MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_IMPL_HPP
+
+namespace mlpack {
+namespace matrix_completion {
+
+MatrixCompletion::MatrixCompletion(const size_t m,
+                                   const size_t n,
+                                   const arma::umat& indices,
+                                   const arma::vec& values,
+                                   const size_t r)
+  : m(m), n(n), indices(indices), values(values),
+    sdp(indices.n_cols, 0, CreateInitialPoint(m, n, r))
+{
+  checkValues();
+  initSdp();
+}
+
+MatrixCompletion::MatrixCompletion(const size_t m,
+                                   const size_t n,
+                                   const arma::umat& indices,
+                                   const arma::vec& values,
+                                   const arma::mat& initialPoint)
+  : m(m), n(n), indices(indices), values(values),
+    sdp(indices.n_cols, 0, initialPoint)
+{
+  checkValues();
+  initSdp();
+}
+
+MatrixCompletion::MatrixCompletion(const size_t m,
+                                   const size_t n,
+                                   const arma::umat& indices,
+                                   const arma::vec& values)
+  : m(m), n(n), indices(indices), values(values),
+    sdp(indices.n_cols, 0, CreateInitialPoint(m, n, DefaultRank(m, n, indices.n_cols)))
+{
+  checkValues();
+  initSdp();
+}
+
+void MatrixCompletion::checkValues()
+{
+  if (indices.n_rows != 2)
+    Log::Fatal << "indices.n_rows != 2" << std::endl;
```

Comment:
Sure. Alternatively, I wonder if arma has a particular type where the # of rows is set to 2 (let the type system enforce this!). 

I fully support `Log::Fatal` returning backtraces. I think it's not too bad if you compile with `-fno-omit-frame-pointer`.